### DISCUSSION TITLE
[Bug]: During upgrade process a dll has not been upgraded and a provider has not been removed

### DISCUSSION THREAD

**Discussion by fablaser** [dnnsoftware/Dnn.Platform#6163] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Hello
I compared a fresh new install of DNN 9.13.04 with my production environment. (same version) I founf that Newtonsoft.Json.dll of my production environment has not been upgraded during years of upgrades (in production environment I have 13.0.1.25517 - that is considered unsecure - while in new install I have 13.0.3.27908.

In production environment I have also:

- Dnn.Modules.ModuleCreator.dll (that can be easly removed uninstalling obsolete "Module Creator" module)
- DotNetNuke.Providers.FiftyOneClientCapabilityProvider.dll and FiftyOne.Foundation.dll that are related to **DotNetNuke Client Capability Provider v9.1.0** that I think has been replaced by **DotNetNuke ASP.NET Client Capability Provider v9.13.4**.

Perhaps future updates should force the Newtonsoft.Json.dll replacement and the removal of  DotNetNuke Client Capability Provider v9.1.0.

Does anyone knows how to remove DotNetNuke Client Capability Provider v9.1.0 provider?

**EDIT: A similar issue with Newtonsoft.Json.dll happened with another DNN installation. In this case the website (created on 2013) is v 13.0.2.27524. I checked the DNN_Platform_9.13.4_Upgrade.zip file and this file is missing... Is it a wanted behavior?**

Thanks

### Steps to reproduce?

Upgrading DNN from older versions

### Affected Versions

9.13.4 (alpha)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mitchelsellers** [comment]
Please be aware that security related concerns should be reported via the official security policy as outlined in this repository.  I have updated the title of this issue to remove the reference to security issue at this time.

I've also temporarily converted this over to a discussion as I'm not sure that we have a true issue here at this time, so a few notes/questions for you.

1. It is not expected that the FiftyOneClientCapabilityProvider is uninstalled, DNN upgrades do not remove older processes.  You should be able to delete this provider under the "Providers" option within the "Extensons" page of the persona bar.
2. With regards to Newtonsoft Json and upgrades.  That DLL is utilized by many third-party vendors, and some of those third-party vendors do things that are not exactly great and can override defaults.  The DNN 9.13.4 upgrade package DOES include the proper version of Newtonsoft JSON it is just included in the /Install/Module folder as it is extracted during upgrade for compatibility sake

I would look at your web.config and look for an Assembly Binding Redirect for Newtonsoft.json and see if any third-party modules are noted as modifying that binding redirect.

**Reply by fablaser** [reply]
> I would look at your web.config and look for an Assembly Binding Redirect for Newtonsoft.json and see if any third-party modules are noted as modifying that binding redirect.

Thank you
How can I privately send you my web.config file?
